code/game/turfs/simulated/floor.dm

```diff
@@ -633,6 +633,18 @@ turf/simulated/floor/proc/update_icon()
 			overlays -= wet_overlay
 			wet_overlay = null
 
+/turf/simulated/floor/attack_construct(mob/user as mob)
+	if(istype(src,/turf/simulated/floor/carpet))	return//carpets are cool
+	if(istype(user,/mob/living/simple_animal/construct/builder) && (get_dist(src,user) <= 3))
+		if((icon_state != "cult")&&(icon_state != "cult-narsie"))
+			var/spell/aoe_turf/conjure/floor/S = locate() in user.spell_list
+			S.perform(user,0,src)
+			if(user.spell_masters && user.spell_masters.len)

```

Comment:
Rather than doing this, you can simply update S.connected_button by doing
`S.connected_button.update_charge(1)`